apollo-backend/src/main/java/io/logz/apollo/notifications/senders/MarkersSender.java

```diff
@@ -0,0 +1,100 @@
+package io.logz.apollo.notifications.senders;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import io.logz.apollo.common.HttpStatus;
+import io.logz.apollo.models.DeploymentMarker;
+import io.logz.apollo.models.DeploymentMarkers;
+import io.logz.apollo.notifications.NotificationTemplateMetadata;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import org.mariadb.jdbc.internal.logging.Logger;
+import org.mariadb.jdbc.internal.logging.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+public class MarkersSender implements NotificationSender {
+    private static final Logger logger = LoggerFactory.getLogger(MarkersSender.class);
+    private final String X_API_TOKEN_HEADER = "X-API-TOKEN";
+    private final String apiToken;
+    private final String createMarkersEndpointUrl;
+
+    public MarkersSender(String notificationJsonConfiguration) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        DeploymentMarkersConfiguration markersConfiguration = mapper.readValue(notificationJsonConfiguration, DeploymentMarkersConfiguration.class);
+        this.apiToken = markersConfiguration.getApiToken();
+        this.createMarkersEndpointUrl = markersConfiguration.getCreateMarkersEndpointUrl();
+    }
+
+    @Override
+    public boolean send(NotificationTemplateMetadata notificationTemplateMetadata) {
+        OkHttpClient httpClient = new OkHttpClient();
```

Comment:
creating OkHttpClient and Request builder could be extracted to constuctor / init function to avoid creating new objects on each send retry

Reply:
won't be that efficient here since it's being created for each new notification, but will do